File: backend/ServiceBooking.Api/Entities/Booking.cs
using ServiceBooking.Api.Enums;

namespace ServiceBooking.Api.Entities;

public class Booking
{
  public Guid Id { get; set; }

  public string BookingCode { get; set; } = null!;

  public Guid CustomerId { get; set; }

  public Guid ServiceId { get; set; }

  public Guid StaffId { get; set; }

  public DateTime StartTime { get; set; }

  public DateTime EndTime { get; set; }

  public BookingStatus Status { get; set; }

  public string? CustomerNote { get; set; }

  public string? CancellationReason { get; set; }

  public DateTime CreatedAt { get; set; }

  public User Customer { get; set; } = null!;

  public Service Service { get; set; } = null!;

  public Staff Staff { get; set; } = null!;
}

File: frontend/types/booking.ts
export type BookingStatus = "Pending" | "Confirmed" | "Completed" | "Cancelled";

export interface AvailableRange {
  startTime: string;
  endTime: string;
}

export interface AvailableSlots {
  serviceId: string;
  staffId: string;
  date: string;
  durationMinutes: number;
  availableRanges: AvailableRange[];
}

export interface CreateBookingRequest {
  serviceId: string;
  staffId: string;
  startTime: string;
  customerNote?: string;
}

export interface Booking {
  id: string;
  bookingCode: string;
  customerId: string;
  customerEmail: string;
  serviceId: string;
  serviceName: string;
  staffId: string;
  staffName: string;
  startTime: string;
  endTime: string;
  status: BookingStatus;
  customerNote?: string | null;
  cancellationReason?: string | null;
  createdAt: string;
}
